### PR TITLE
SMath Wrapper refactoring

### PR DESCRIPTION
**Pull request by dcprojects**
PR for #1355
@mikekaganski

### CODE REVIEW

**Comment by ibell** [comment]
Hey @mikekaganski can you do a code review on this?

File: wrappers/SMath/coolprop_wrapper/Functions/CoolProp_HAProps.cs [hunk 1]
```diff
@@ -3,136 +3,136 @@
...
-    }
+            return new TermInfo(inf.Text,
+                                inf.Type,
+                                funcInfo,
+                                FunctionSections.Unknown,
+                                true,
+                                argsInfos);
+        }
 
-    bool IFunction.ExpressionEvaluation(Term root, Term[][] args, ref SMath.Math.Store context, ref Term[] result)
```
Comment (mikekaganski) [inline wrappers/SMath/coolprop_wrapper/Functions/CoolProp_HAProps.cs]: Is it OK that we pass context by value? Won't be here substantial overhead?

Comment (dcprojects) [inline wrappers/SMath/coolprop_wrapper/Functions/CoolProp_HAProps.cs]: not sure if this is the only reason behind this change, but context is a reference type, thus not using ref should be an issue only if one wants to initialize a new instance (that shouldn't happens plugin-side);

Comment (mikekaganski) [inline wrappers/SMath/coolprop_wrapper/Functions/CoolProp_HAProps.cs]: Thank you! I'm fine with this.

**Comment by ibell** [comment]
Thanks guys

**Pull request merged**